parlai/crowdsourcing/tasks/model_chat/analysis/compile_results.py

```diff
@@ -283,8 +283,10 @@ def compile_results(self) -> pd.DataFrame:
                         f'Got long dialogue of {len(data["dialog"])} utterances, hit id: {info_dict["hit_id"]}, model_nickname: {model_nickname}.'
                     )
 
-                dialog_has_problems = False
+                if self.use_problem_buckets:
+                    dialog_has_problems = False
```

Comment:
As a note, I'm finding the numerous portions that are gated by `use_problem_buckets` throughout to be somewhat hard to follow. In this case, we're even creating variable to be used by other scopes only if the attribute is set. 

It could be that the overall `compile_results()` code is rather monolithic, so I have a hard time seeing where it could be broken down or have the `self.use_problem_buckets` idea pulled out.

Not necessarily blocking, especially if you believe this to be the only type of branching that could occur in these analysis scripts, but worth discussing.

Reply:
Yes, I agree that the gating of functionality with `self.use_problem_buckets` adds complexity to an already very complex method, `.compile_results()`: it's not ideal to do this, but in my mind I feel like this is the most straightforward way to achieve this without completely rewriting the method.

Soon I'll be having a larger discussion about how we want to structure all analysis code going forward, so my hope is that, in the medium-term, this code will be overhauled more thoroughly in order to make it less monolithic; thus, I see this PR as a stopgap solution to provide needed functionality. Happy to discuss if you think there is a better stopgap solution for this :)

Reply:
If you have a brainstorming session planned for that overhaul, then I don't see a need to delay this step.